modules/st2-panel/panel.component.js

```diff
@@ -370,22 +370,37 @@ export class DetailsCriteriaLine extends React.Component {
   static propTypes = {
     name: PropTypes.string,
     type: PropTypes.string,
-    pattern: PropTypes.string,
+    pattern: PropTypes.any,
+    condition: PropTypes.string,
   }
 
   render() {
-    const { name, type, pattern } = this.props;
+    const { name, type, pattern, condition } = this.props;
+
+    const search = type !== 'search' ? null : {
+      condition,
+      pattern: pattern['item.data'].pattern,
+      type: pattern['item.data'].type,
```

Comment:
Looks like there was a misunderstanding here and so this PR is always expecting to have `item.data` in the rule pattern.

I now see this is coming from the example: 
https://github.com/bitovi-stackstorm-exchange/st2-ui-workflow-tests/blob/main/rules/validate-alert.yaml#L16

But based on https://github.com/bitovi-stackstorm-exchange/st2-ui-workflow-tests/blob/4f576228aad17652ceb47b1f444f2624f4c116c8/rules/validate-alert.yaml#L21

instead of `item.data` could be `item.foo`, `item.bar` and so on, as that refers to the dictionary item.

And so based on implementation here, this works: 
```yaml
---
name: test_search1_ok
pack: st2
enabled: true

trigger:
  type: "core.st2.webhook"
  parameters:
    url: "webhook-alert/handle-alert"

criteria:
  trigger.body.alert:
    type: "search"
    condition: any
    pattern:
      item.data: # works
        type: "equals"
        pattern: "foo"

action:
  ref: "core.echo"
  parameters:
    message: "{{ trigger.body }}"
```

While this rule doesn't:
```yaml
---
name: test_search1_err
pack: st2
enabled: true

trigger:
  type: "core.st2.webhook"
  parameters:
    url: "webhook-alert/handle-alert"

criteria:
  trigger.body.alert:
    type: "search"
    condition: any
    pattern:
      item.time: # changed, it's just a dict item
        type: "equals"
        pattern: "foo"

action:
  ref: "core.echo"
  parameters:
    message: "{{ trigger.body }}"

```